### PR TITLE
Graphics context

### PR DESCRIPTION
**Pull request by hrydgard**
Abstract away things like swapbuffers etc before adding even more backends.

Needed to prevent "switch (gpu)" clutter all over the codebase later.

Does not go all the way yet, goal would be a common render loop between platforms but not there yet.

### CODE REVIEW

File: headless/WindowsHeadlessHost.cpp [hunk 1]
```diff
@@ -169,13 +169,14 @@ bool WindowsHeadlessHost::InitGraphics(std::string *error_message)
...
 
-	GL_SwapInterval(0);
+	// GL_SwapInterval(0);
 
 	glewInit();
 	CheckGLExtensions();
 
 	LoadNativeAssets();
 
+	*graphicsContext = new DummyGraphicsContext();
```
Comment (unknownbrackets) [inline headless/WindowsHeadlessHost.cpp]: Does headless still properly do the rendering?

-[Unknown]

File: ios/ViewController.mm [hunk 2]
```diff
@@ -247,7 +247,7 @@ - (void)glkView:(GLKView *)view drawInRect:(CGRect)rect
 		EndInputState(&input_state);
 	}
 
-	NativeRender();
+	NativeRender(NULL);
```
Comment (unknownbrackets) [inline ios/ViewController.mm]: Is NULL okay here?  It doesn't need a dummy?

File: lang [hunk 3]
```diff
@@ -1 +1 @@
-Subproject commit 9900b0978a3e51dfe8d58cd86c6a58d3a0135580
+Subproject commit 84fd93497b9718a89ac04edfe9636b0f90943505
```
Comment (unknownbrackets) [inline lang]: This reverts back to an older version.

-[Unknown]

Comment (hrydgard) [inline lang]: Yeah, needs more work before review.

**Comment by hrydgard** [comment]
Hmm, I can't decide if this is the right way to go or not... We already have "Host" which does kind of the same thing although it's way too unfocused - does a little bit of everything.

**Comment by unknownbrackets** [comment]
Like the input_state cleanup, yay finally.  Personally I think this is a better direction, as you said on IRC.  I mean, yeah, stuff is gonna break but only temporarily, 

-[Unknown]

**Comment by unknownbrackets** [comment]
Only Qt left: https://github.com/hrydgard/ppsspp/blob/5ff82240755734f09946e336ef36bb4bbe64e9fd/ext/native/base/QtMain.h#L186

-[Unknown]

**Comment by hrydgard** [comment]
OK, that should be that. Let's see what travis says..

File: Windows/GPU/WindowsGLContext.cpp [hunk 4]
```diff
@@ -392,3 +382,14 @@ void GL_Shutdown() {
...
 	hWnd = NULL;
 }
+
+void WindowsGLContext::Resize() {
+}
+
+Thin3DContext *WindowsGLContext::CreateThin3DContext() {
+	Thin3DContext *ctx = T3DCreateGLContext();
+	if (ctx) {
+		CheckGLExtensions();
```
Comment (unknownbrackets) [inline Windows/GPU/WindowsGLContext.cpp]: This is only happening on Windows.  I guess that's #8387.

-[Unknown]

**Pull request merged**